### ISSUE TITLE
Rest API: the authentication of privileges 

### ISSUE THREAD

**Issue by leefrank9527** [WebCuratorTool/webcurator#131]
Currently the default role of Privilege is used for debug inside RestApiAuthFilter.java. That should be changed to use the actual role. There are 2 options to set the role of request.

- Option 1: Set the role in the http header from the client side. But it's not secure, the attackers can set the privileged role for the other roles. To resolve this, the server should validate the consistence of the role and path, method.
- Option 2: Calculate the role according to the request path and method.

Both options need to be implemented from the server side. The option 2 seems friendly to the client. The client don't have to know and understand the concept of role.

**Comment by hannakoppelaar** [comment]
Yes, there's currently no authorization in the API. There's a FIXME in RestApiAuthFilter.java ;) I agree, this can only be done server-side, without relying on client parameters.

**Comment by hannakoppelaar** [comment]
I've implemented almost all roles and scopes to the extent possible, given that not all APIs have been realized yet (branch features/ui-rebuild-131). There are a few questions left:

* The difference between the roles MANAGE_WEB_HARVESTER and MANAGE_TARGET_INSTANCES is not entirely clear. The existing code mostly treats them as equivalent, except in the case of notifications, where only MANAGE_WEB_HARVESTER is used. Maybe we can drop one of them? 

* The concept of profile levels is confusing. It works as follows. A profile has a level attribute that can have values 1, 2 and 3. To these values correspond three roles: SET_HARVEST_PROFILE_LV[123]. Now, in order to be able to add a profile with level m to a target, the user needs to have a role SET_HARVEST_PROFILE_LV<n> where n >= m. Exception to this rule: if the profile is the default profile it can always be added to the target. Is this actually being used? Do we want to keep it?

* The way the role VIEW_PROFILES is used in the old UI, boils down to silently filtering the profiles that the user is not allowed to see: if a user does not have the role VIEW_PROFILES or the role MANAGE_PROFILES, only the profiles belonging to the user's agency are shown. Since the filtering of profiles is being done silently, this might become a source of hard to spot errors. Maybe we should add something to the output of GET /profiles to indicate which profiles are being withheld from the response?
